bindings/jupyroot/python/JupyROOT/helpers/utils.py

```diff
@@ -242,6 +242,49 @@ def invokeAclic(cell):
     else:
         processCppCode(".L %s+" %fileName)
 
+def produceCanvasJsCode(canvas):
+   # Add extra primitives to canvas with custom colors, palette, gStyle
+   prim = canvas.GetListOfPrimitives()
+
+   style = ROOT.gStyle
+   colors = ROOT.gROOT.GetListOfColors()
+   palette = None
+
+   # always provide gStyle object
+   if prim.FindObject(style):
+      style = None
+   else:
+      prim.Add(style)
+
+   cnt = 0
+   for n in range(colors.GetLast()+1):
+      if colors.At(n): cnt = cnt+1
+        
+   # add all colors if there are more than 598 colors defined
+   if cnt < 599 or prim.FindObject(colors):
+      colors = None
+   else:
+      prim.Add(colors)
+
+   if colors:
+      pal = ROOT.TColor.GetPalette()
+      palette = ROOT.TObjArray()
+      palette.SetName("CurrentColorPalette")
+      for i in range(pal.GetSize()):
+         palette.Add(colors.At(pal[i]))
+      prim.Add(palette)
+
+   ROOT.TColor.DefinedColors()
+    
+   canvas_json = ROOT.TBufferJSON.ConvertToJSON(canvas, 3)
+
+   # Cleanup primitives after conversion
+   if style: prim.Remove(style)
```

Comment:
As I commented earlier, this should be `if style is not None`, since you can't rely on the boolean value of style (perhaps ROOT.gStyle could reimplement its boolean behaviour in Python). Same for `colors` and `palette` below.